segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -1040,24 +1039,22 @@ public void readTableEntriesDelta(WireCommands.ReadTableEntriesDelta readTableEn
 
     }
 
-    private void truncateIndexSegment(String segment, long offset) {
+    private CompletableFuture<Void> truncateIndexSegment(String segment, long offset) {
         String indexSegment = getIndexSegmentName(segment);
         try {
             long indexSegmentOffset = IndexRequestProcessor.locateOffsetForIndexSegment(segmentStore, segment, offset, false);
-            log.debug("Truncating index segment {} at offset {}.", indexSegment, indexSegmentOffset);
+            log.info("Truncating index segment {} at offset {}.", indexSegment, indexSegmentOffset);
             if (indexSegmentOffset == 0) {
-                return;
+                return CompletableFuture.completedFuture(null);
             }
-            segmentStore.truncateStreamSegment(indexSegment, indexSegmentOffset, TIMEOUT)
-                    .whenComplete((v, exp) -> {
-                        if (exp != null) {
-                            log.warn("Failed to truncate index segment {} at offset {} due to", indexSegment, indexSegmentOffset, exp);
-                        } else {
-                            log.info("Successfully truncated index segment {} at offset {}", indexSegment, indexSegmentOffset);
-                        }
-                    });
-        } catch (Exception e) {
-            log.warn("Unable to locate offset for index segment {}  for offset {} due to ", indexSegment, offset, e);
+            return segmentStore.truncateStreamSegment(indexSegment, indexSegmentOffset, TIMEOUT);
+        } catch (Exception ex) {
+            if (ex instanceof IllegalArgumentException || ex instanceof IllegalStateException || ex instanceof IndexRequestProcessor.SegmentTruncatedException) {
```

Comment:
Please add test coverage for both the the if an else cases here

Reply:
Added